src/guides/language/evolution.md

```diff
@@ -37,6 +40,50 @@ environment:
 
 ## Changes in each release
 
+### Dart 3.2
+_Released 15 November 2023_
+| [Dart 3.2 announcement](https://medium.com/dartlang/)
+
+Dart 3.2 added enhancements to flow analysis, including:
+
+* Expanded [type promotion](/null-safety/understanding-null-safety#type-promotion-on-null-checks)
+  to work on private final fields. Previously only available for
+  local variables and parameters, now private final fields can promote to
+  non-nullable types through null checks and `is` tests. For example,
+  the following code is now sound:
+
+  ```dart
+  class Example {
+  final int? _privateField;
+  Example1(this._privateField);
+
+  f() {
+    if (_privateField != null) {
+      // _privateField has now been promoted; you can use it without
+      // null checking it.
+      int i = _privateField; // OK
+      }
+    }
+  }
+
+  // Private field promotions also work from outside of the class:
+  f(Example1 x) {
+    if (x._privateField != null) {
+      int i = x._privateField; // OK
+    }
+  }
```

Comment:
I'm assuming it wasn't intended for there to be a mix of `Example` and `Example1`.

Fixes to that and formatting/indentation:

```suggestion
  class Example {
    final int? _privateField;
  
    Example(this._privateField);
  
    void f() {
      if (_privateField != null) {
        // _privateField has now been promoted; you can use it without
        // null checking it.
        int i = _privateField; // OK
      }
    }
  }
  
  // Private field promotions also work from outside of the class:
  void f(Example x) {
    if (x._privateField != null) {
      int i = x._privateField; // OK
    }
  }
```